diff -ur click-1.4.1-orig/doc/Makefile.in click-1.4.1/doc/Makefile.in
--- click-1.4.1-orig/doc/Makefile.in	2004-07-02 21:51:26.000000000 -0600
+++ click-1.4.1/doc/Makefile.in	2004-08-11 16:24:15.000000000 -0600
@@ -30,36 +30,36 @@
 	  && CONFIG_FILES=$(subdir)/$@ CONFIG_HEADERS= $(SHELL) ./config.status
 
 install: Makefile install-man install-info
-	$(mkinstalldirs) $(bindir)
-	$(INSTALL) $(srcdir)/click-elem2man $(bindir)/click-elem2man
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
+	$(INSTALL) $(srcdir)/click-elem2man $(DESTDIR)/$(bindir)/click-elem2man
 install-man: $(srcdir)/testie.1 Makefile
-	$(mkinstalldirs) $(mandir)/man1 $(mandir)/man3 $(mandir)/man5 $(mandir)/man7 $(mandir)/man8 $(mandir)/mann
-	$(INSTALL_DATA) $(srcdir)/click.1 $(mandir)/man1/click.1
-	$(INSTALL_DATA) $(srcdir)/click.5 $(mandir)/man5/click.5
-	$(INSTALL_DATA) $(srcdir)/click.o.8 $(mandir)/man8/click.o.8
-	$(INSTALL_DATA) $(srcdir)/click-align.1 $(mandir)/man1/click-align.1
-	$(INSTALL_DATA) $(srcdir)/click-combine.1 $(mandir)/man1/click-combine.1
-	$(INSTALL_DATA) $(srcdir)/click-devirtualize.1 $(mandir)/man1/click-devirtualize.1
-	$(INSTALL_DATA) $(srcdir)/click-fastclassifier.1 $(mandir)/man1/click-fastclassifier.1
-	$(INSTALL_DATA) $(srcdir)/click-flatten.1 $(mandir)/man1/click-flatten.1
-	$(INSTALL_DATA) $(srcdir)/click-install.1 $(mandir)/man1/click-install.1
-	$(INSTALL_DATA) $(srcdir)/click-mkmindriver.1 $(mandir)/man1/click-mkmindriver.1
-	$(INSTALL_DATA) $(srcdir)/click-pretty.1 $(mandir)/man1/click-pretty.1
-	$(INSTALL_DATA) $(srcdir)/click-uncombine.1 $(mandir)/man1/click-uncombine.1
-	$(INSTALL_DATA) $(srcdir)/click-undead.1 $(mandir)/man1/click-undead.1
-	$(INSTALL_DATA) $(srcdir)/click-uninstall.1 $(mandir)/man1/click-uninstall.1
-	$(INSTALL_DATA) $(srcdir)/click-xform.1 $(mandir)/man1/click-xform.1
-	$(INSTALL_DATA) $(srcdir)/elementdoc.7 $(mandir)/man7/elementdoc.7
-	$(INSTALL_DATA) $(srcdir)/testie.1 $(mandir)/man1/testie.1
-	echo @TARGETS@ @element_groups@ $(EXTRA_PROVIDES) | $(top_builddir)/click-buildtool findelem -r "@TARGETS@ $(EXTRA_PROVIDES)" -p $(top_srcdir) | $(PERL) $(srcdir)/click-elem2man -l -d $(mandir)/mann -p DEFAULT -f -
+	$(mkinstalldirs) $(DESTDIR)/$(mandir)/man1 $(DESTDIR)/$(mandir)/man3 $(DESTDIR)/$(mandir)/man5 $(DESTDIR)/$(mandir)/man7 $(DESTDIR)/$(mandir)/man8 $(DESTDIR)/$(mandir)/mann
+	$(INSTALL_DATA) $(srcdir)/click.1 $(DESTDIR)/$(mandir)/man1/click.1
+	$(INSTALL_DATA) $(srcdir)/click.5 $(DESTDIR)/$(mandir)/man5/click.5
+	$(INSTALL_DATA) $(srcdir)/click.o.8 $(DESTDIR)/$(mandir)/man8/click.o.8
+	$(INSTALL_DATA) $(srcdir)/click-align.1 $(DESTDIR)/$(mandir)/man1/click-align.1
+	$(INSTALL_DATA) $(srcdir)/click-combine.1 $(DESTDIR)/$(mandir)/man1/click-combine.1
+	$(INSTALL_DATA) $(srcdir)/click-devirtualize.1 $(DESTDIR)/$(mandir)/man1/click-devirtualize.1
+	$(INSTALL_DATA) $(srcdir)/click-fastclassifier.1 $(DESTDIR)/$(mandir)/man1/click-fastclassifier.1
+	$(INSTALL_DATA) $(srcdir)/click-flatten.1 $(DESTDIR)/$(mandir)/man1/click-flatten.1
+	$(INSTALL_DATA) $(srcdir)/click-install.1 $(DESTDIR)/$(mandir)/man1/click-install.1
+	$(INSTALL_DATA) $(srcdir)/click-mkmindriver.1 $(DESTDIR)/$(mandir)/man1/click-mkmindriver.1
+	$(INSTALL_DATA) $(srcdir)/click-pretty.1 $(DESTDIR)/$(mandir)/man1/click-pretty.1
+	$(INSTALL_DATA) $(srcdir)/click-uncombine.1 $(DESTDIR)/$(mandir)/man1/click-uncombine.1
+	$(INSTALL_DATA) $(srcdir)/click-undead.1 $(DESTDIR)/$(mandir)/man1/click-undead.1
+	$(INSTALL_DATA) $(srcdir)/click-uninstall.1 $(DESTDIR)/$(mandir)/man1/click-uninstall.1
+	$(INSTALL_DATA) $(srcdir)/click-xform.1 $(DESTDIR)/$(mandir)/man1/click-xform.1
+	$(INSTALL_DATA) $(srcdir)/elementdoc.7 $(DESTDIR)/$(mandir)/man7/elementdoc.7
+	$(INSTALL_DATA) $(srcdir)/testie.1 $(DESTDIR)/$(mandir)/man1/testie.1
+	echo @TARGETS@ @element_groups@ $(EXTRA_PROVIDES) | $(top_builddir)/click-buildtool findelem -r "@TARGETS@ $(EXTRA_PROVIDES)" -p $(top_srcdir) | $(PERL) $(srcdir)/click-elem2man -l -d $(DESTDIR)/$(mandir)/mann -p DEFAULT -f -
 install-info: Makefile $(srcdir)/click.info
-	$(mkinstalldirs) $(infodir)
+	$(mkinstalldirs) $(DESTDIR)/$(infodir)
 	@for f in `cd $(srcdir) && echo click.info click.info-[0-9] click.info-[0-9][0-9]`; do \
 	  if test -f $(srcdir)/$$f; then \
-	    echo "$(INSTALL_DATA) $(srcdir)/$$f $(infodir)/$$f"; \
-	    $(INSTALL_DATA) $(srcdir)/$$f $(infodir)/$$f || exit 1; fi; \
+	    echo "$(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)/$(infodir)/$$f"; \
+	    $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)/$(infodir)/$$f || exit 1; fi; \
 	done
-	-$(INSTALL_INFO) --info-dir=$(infodir) $(infodir)/click.info 2>&1 >/dev/null
+	-$(INSTALL_INFO) --info-dir=$(infodir) $(DESTDIR)/$(infodir)/click.info 2>&1 >/dev/null
 
 uninstall: uninstall-man uninstall-info
 	/bin/rm -f $(bindir)/click-elem2man
diff -ur click-1.4.1-orig/Makefile.in click-1.4.1/Makefile.in
--- click-1.4.1-orig/Makefile.in	2004-07-07 01:03:05.000000000 -0600
+++ click-1.4.1/Makefile.in	2004-08-11 16:02:18.000000000 -0600
@@ -58,16 +58,16 @@
 	@for d in $(ALL_TARGETS); do (cd $$d && $(MAKE) install) || exit 1; done
 	@$(MAKE) install-local install-doc install-local-include
 install-local: elementmap.xml click-buildtool click-compile
-	$(mkinstalldirs) $(bindir)
-	$(INSTALL_IF_CHANGED) click-buildtool $(bindir)/click-buildtool
-	$(INSTALL_IF_CHANGED) click-compile $(bindir)/click-compile
-	$(INSTALL_IF_CHANGED) $(srcdir)/click-mkelemmap $(bindir)/click-mkelemmap
-	$(INSTALL_IF_CHANGED) $(top_srcdir)/test/testie $(bindir)/testie
-	$(mkinstalldirs) $(clickdatadir)
-	$(INSTALL_DATA) elementmap.xml $(clickdatadir)/elementmap.xml
-	(cd $(top_srcdir); pwd) > $(clickdatadir)/srcdir
-	/bin/rm -rf $(clickdatadir)/src
-	/bin/ln -s `cd $(top_srcdir); pwd` $(clickdatadir)/src
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
+	$(INSTALL_IF_CHANGED) click-buildtool $(DESTDIR)/$(bindir)/click-buildtool
+	$(INSTALL_IF_CHANGED) click-compile $(DESTDIR)/$(bindir)/click-compile
+	$(INSTALL_IF_CHANGED) $(srcdir)/click-mkelemmap $(DESTDIR)/$(bindir)/click-mkelemmap
+	$(INSTALL_IF_CHANGED) $(top_srcdir)/test/testie $(DESTDIR)/$(bindir)/testie
+	$(mkinstalldirs) $(DESTDIR)/$(clickdatadir)
+	$(INSTALL_DATA) elementmap.xml $(DESTDIR)/$(clickdatadir)/elementmap.xml
+	(cd $(top_srcdir); pwd) > $(DESTDIR)/$(clickdatadir)/srcdir
+	/bin/rm -rf $(DESTDIR)/$(clickdatadir)/src
+	/bin/ln -s `cd $(top_srcdir); pwd` $(DESTDIR)/$(clickdatadir)/src
 install-doc:
 	@cd doc && $(MAKE) install
 install-info:
@@ -75,15 +75,15 @@
 install-man:
 	@-for d in $(ALL_TARGETS) doc; do (cd $$d && $(MAKE) install-man); done
 install-local-include: stamp-h
-	$(mkinstalldirs) $(clickincludedir)
-	$(INSTALL_DATA_IF_CHANGED) $(srcdir)/include/click/*.h $(clickincludedir)
-	$(INSTALL_DATA_IF_CHANGED) $(srcdir)/include/click/*.hh $(clickincludedir)
-	$(INSTALL_DATA_IF_CHANGED) $(srcdir)/include/click/*.cc $(clickincludedir)
-	$(INSTALL_DATA_IF_CHANGED) $(top_builddir)/include/click/*.h $(clickincludedir)
-	$(mkinstalldirs) $(clickincludedir)/standard
-	$(INSTALL_DATA_IF_CHANGED) $(srcdir)/include/click/standard/*.hh $(clickincludedir)/standard
-	$(mkinstalldirs) $(netincludedir)
-	$(INSTALL_DATA_IF_CHANGED) $(srcdir)/include/clicknet/*.h $(netincludedir)
+	$(mkinstalldirs) $(DESTDIR)/$(clickincludedir)
+	$(INSTALL_DATA_IF_CHANGED) $(srcdir)/include/click/*.h $(DESTDIR)/$(clickincludedir)
+	$(INSTALL_DATA_IF_CHANGED) $(srcdir)/include/click/*.hh $(DESTDIR)/$(clickincludedir)
+	$(INSTALL_DATA_IF_CHANGED) $(srcdir)/include/click/*.cc $(DESTDIR)/$(clickincludedir)
+	$(INSTALL_DATA_IF_CHANGED) $(top_builddir)/include/click/*.h $(DESTDIR)/$(clickincludedir)
+	$(mkinstalldirs) $(DESTDIR)/$(clickincludedir)/standard
+	$(INSTALL_DATA_IF_CHANGED) $(srcdir)/include/click/standard/*.hh $(DESTDIR)/$(clickincludedir)/standard
+	$(mkinstalldirs) $(DESTDIR)/$(netincludedir)
+	$(INSTALL_DATA_IF_CHANGED) $(srcdir)/include/clicknet/*.h $(DESTDIR)/$(netincludedir)
 install-include: install-local-include
 	@cd tools && $(MAKE) install-include
 
diff -ur click-1.4.1-orig/ns/Makefile.in click-1.4.1/ns/Makefile.in
--- click-1.4.1-orig/ns/Makefile.in	2004-06-17 23:33:53.000000000 -0600
+++ click-1.4.1/ns/Makefile.in	2004-08-11 16:15:35.000000000 -0600
@@ -106,8 +106,8 @@
 endif
 
 install: libnsclick.a
-	$(mkinstalldirs) $(libdir)
-	$(INSTALL) libnsclick.a $(libdir)/libnsclick.a
+	$(mkinstalldirs) $(DESTDIR)/$(libdir)
+	$(INSTALL) libnsclick.a $(DESTDIR)/$(libdir)/libnsclick.a
 uninstall:
 	/bin/rm -f $(libdir)/libnsclick.a
 
diff -ur click-1.4.1-orig/tools/click2xml/Makefile.in click-1.4.1/tools/click2xml/Makefile.in
--- click-1.4.1-orig/tools/click2xml/Makefile.in	2004-01-14 18:37:59.000000000 -0700
+++ click-1.4.1/tools/click2xml/Makefile.in	2004-08-11 16:20:46.000000000 -0600
@@ -85,9 +85,9 @@
 
 install: $(LIBCLICKTOOL) install-local
 install-local: all-local
-	$(mkinstalldirs) $(bindir)
-	$(INSTALL) click2xml $(bindir)/click2xml
-	test -z "@XML2CLICK@" || $(INSTALL) xml2click $(bindir)/xml2click
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
+	$(INSTALL) click2xml $(DESTDIR)/$(bindir)/click2xml
+	test -z "@XML2CLICK@" || $(INSTALL) xml2click $(DESTDIR)/$(bindir)/xml2click
 uninstall:
 	/bin/rm -f $(bindir)/click2xml
 	test -z "@XML2CLICK@" || /bin/rm -f $(bindir)/xml2click
diff -ur click-1.4.1-orig/tools/click-align/Makefile.in click-1.4.1/tools/click-align/Makefile.in
--- click-1.4.1-orig/tools/click-align/Makefile.in	2004-01-14 18:37:59.000000000 -0700
+++ click-1.4.1/tools/click-align/Makefile.in	2004-08-11 16:17:17.000000000 -0600
@@ -81,8 +81,8 @@
 
 install: $(LIBCLICKTOOL) install-local
 install-local: all-local
-	$(mkinstalldirs) $(bindir)
-	$(INSTALL) click-align $(bindir)/click-align
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
+	$(INSTALL) click-align $(DESTDIR)/$(bindir)/click-align
 uninstall:
 	/bin/rm -f $(bindir)/click-align
 
diff -ur click-1.4.1-orig/tools/click-check/Makefile.in click-1.4.1/tools/click-check/Makefile.in
--- click-1.4.1-orig/tools/click-check/Makefile.in	2004-01-14 18:37:59.000000000 -0700
+++ click-1.4.1/tools/click-check/Makefile.in	2004-08-11 16:17:25.000000000 -0600
@@ -81,8 +81,8 @@
 
 install: $(LIBCLICKTOOL) install-local
 install-local: all-local
-	$(mkinstalldirs) $(bindir)
-	$(INSTALL) click-check $(bindir)/click-check
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
+	$(INSTALL) click-check $(DESTDIR)/$(bindir)/click-check
 uninstall:
 	/bin/rm -f $(bindir)/click-check
 
diff -ur click-1.4.1-orig/tools/click-combine/Makefile.in click-1.4.1/tools/click-combine/Makefile.in
--- click-1.4.1-orig/tools/click-combine/Makefile.in	2004-01-14 18:37:59.000000000 -0700
+++ click-1.4.1/tools/click-combine/Makefile.in	2004-08-11 16:17:36.000000000 -0600
@@ -85,9 +85,9 @@
 
 install: $(LIBCLICKTOOL) install-local
 install-local: all-local
-	$(mkinstalldirs) $(bindir)
-	$(INSTALL) click-combine $(bindir)/click-combine
-	$(INSTALL) click-uncombine $(bindir)/click-uncombine
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
+	$(INSTALL) click-combine $(DESTDIR)/$(bindir)/click-combine
+	$(INSTALL) click-uncombine $(DESTDIR)/$(bindir)/click-uncombine
 uninstall:
 	/bin/rm -f $(bindir)/click-combine $(bindir)/click-uncombine
 
diff -ur click-1.4.1-orig/tools/click-devirtualize/Makefile.in click-1.4.1/tools/click-devirtualize/Makefile.in
--- click-1.4.1-orig/tools/click-devirtualize/Makefile.in	2004-01-14 18:37:59.000000000 -0700
+++ click-1.4.1/tools/click-devirtualize/Makefile.in	2004-08-11 16:17:43.000000000 -0600
@@ -81,8 +81,8 @@
 
 install: $(LIBCLICKTOOL) install-local
 install-local: all-local
-	$(mkinstalldirs) $(bindir)
-	$(INSTALL) click-devirtualize $(bindir)/click-devirtualize
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
+	$(INSTALL) click-devirtualize $(DESTDIR)/$(bindir)/click-devirtualize
 uninstall:
 	/bin/rm -f $(bindir)/click-devirtualize
 
diff -ur click-1.4.1-orig/tools/click-fastclassifier/Makefile.in click-1.4.1/tools/click-fastclassifier/Makefile.in
--- click-1.4.1-orig/tools/click-fastclassifier/Makefile.in	2004-01-14 18:37:59.000000000 -0700
+++ click-1.4.1/tools/click-fastclassifier/Makefile.in	2004-08-11 16:17:54.000000000 -0600
@@ -81,8 +81,8 @@
 
 install: $(LIBCLICKTOOL) install-local
 install-local: all-local
-	$(mkinstalldirs) $(bindir)
-	$(INSTALL) click-fastclassifier $(bindir)/click-fastclassifier
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
+	$(INSTALL) click-fastclassifier $(DESTDIR)/$(bindir)/click-fastclassifier
 uninstall:
 	/bin/rm -f $(bindir)/click-fastclassifier
 
diff -ur click-1.4.1-orig/tools/click-flatten/Makefile.in click-1.4.1/tools/click-flatten/Makefile.in
--- click-1.4.1-orig/tools/click-flatten/Makefile.in	2004-01-14 18:37:59.000000000 -0700
+++ click-1.4.1/tools/click-flatten/Makefile.in	2004-08-11 16:18:01.000000000 -0600
@@ -81,8 +81,8 @@
 
 install: $(LIBCLICKTOOL) install-local
 install-local: all-local
-	$(mkinstalldirs) $(bindir)
-	$(INSTALL) click-flatten $(bindir)/click-flatten
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
+	$(INSTALL) click-flatten $(DESTDIR)/$(bindir)/click-flatten
 uninstall:
 	/bin/rm -f $(bindir)/click-flatten
 
diff -ur click-1.4.1-orig/tools/click-install/Makefile.in click-1.4.1/tools/click-install/Makefile.in
--- click-1.4.1-orig/tools/click-install/Makefile.in	2004-01-14 18:37:59.000000000 -0700
+++ click-1.4.1/tools/click-install/Makefile.in	2004-08-11 16:18:24.000000000 -0600
@@ -85,9 +85,9 @@
 
 install: $(LIBCLICKTOOL) install-local
 install-local: all-local
-	$(mkinstalldirs) $(sbindir)
-	$(INSTALL) click-install $(sbindir)/click-install
-	$(INSTALL) click-uninstall $(sbindir)/click-uninstall
+	$(mkinstalldirs) $(DESTDIR)/$(sbindir)
+	$(INSTALL) click-install $(DESTDIR)/$(sbindir)/click-install
+	$(INSTALL) click-uninstall $(DESTDIR)/$(sbindir)/click-uninstall
 install-man:
 	/bin/rm -f $(sbindir)/click-install $(sbindir)/click-uninstall
 
diff -ur click-1.4.1-orig/tools/click-mkmindriver/Makefile.in click-1.4.1/tools/click-mkmindriver/Makefile.in
--- click-1.4.1-orig/tools/click-mkmindriver/Makefile.in	2004-01-14 18:37:59.000000000 -0700
+++ click-1.4.1/tools/click-mkmindriver/Makefile.in	2004-08-11 16:19:10.000000000 -0600
@@ -81,8 +81,8 @@
 
 install: $(LIBCLICKTOOL) install-local
 install-local: all-local
-	$(mkinstalldirs) $(bindir)
-	$(INSTALL) click-mkmindriver $(bindir)/click-mkmindriver
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
+	$(INSTALL) click-mkmindriver $(DESTDIR)/$(bindir)/click-mkmindriver
 uninstall:
 	/bin/rm -f $(bindir)/click-mkmindriver
 
diff -ur click-1.4.1-orig/tools/click-pretty/Makefile.in click-1.4.1/tools/click-pretty/Makefile.in
--- click-1.4.1-orig/tools/click-pretty/Makefile.in	2004-01-14 18:37:59.000000000 -0700
+++ click-1.4.1/tools/click-pretty/Makefile.in	2004-08-11 16:19:18.000000000 -0600
@@ -81,8 +81,8 @@
 
 install: $(LIBCLICKTOOL) install-local
 install-local: all-local
-	$(mkinstalldirs) $(bindir)
-	$(INSTALL) click-pretty $(bindir)/click-pretty
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
+	$(INSTALL) click-pretty $(DESTDIR)/$(bindir)/click-pretty
 uninstall:
 	/bin/rm -f $(bindir)/click-pretty
 
diff -ur click-1.4.1-orig/tools/click-undead/Makefile.in click-1.4.1/tools/click-undead/Makefile.in
--- click-1.4.1-orig/tools/click-undead/Makefile.in	2004-01-14 18:37:59.000000000 -0700
+++ click-1.4.1/tools/click-undead/Makefile.in	2004-08-11 16:19:29.000000000 -0600
@@ -81,8 +81,8 @@
 
 install: $(LIBCLICKTOOL) install-local
 install-local: all-local
-	$(mkinstalldirs) $(bindir)
-	$(INSTALL) click-undead $(bindir)/click-undead
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
+	$(INSTALL) click-undead $(DESTDIR)/$(bindir)/click-undead
 uninstall:
 	/bin/rm -f $(bindir)/click-undead
 
diff -ur click-1.4.1-orig/tools/click-xform/Makefile.in click-1.4.1/tools/click-xform/Makefile.in
--- click-1.4.1-orig/tools/click-xform/Makefile.in	2004-01-14 18:37:59.000000000 -0700
+++ click-1.4.1/tools/click-xform/Makefile.in	2004-08-11 16:19:39.000000000 -0600
@@ -81,8 +81,8 @@
 
 install: $(LIBCLICKTOOL) install-local
 install-local: all-local
-	$(mkinstalldirs) $(bindir)
-	$(INSTALL) click-xform $(bindir)/click-xform
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
+	$(INSTALL) click-xform $(DESTDIR)/$(bindir)/click-xform
 uninstall:
 	/bin/rm -f $(bindir)/click-xform
 
diff -ur click-1.4.1-orig/tools/lib/Makefile.in click-1.4.1/tools/lib/Makefile.in
--- click-1.4.1-orig/tools/lib/Makefile.in	2004-04-17 10:18:44.000000000 -0600
+++ click-1.4.1/tools/lib/Makefile.in	2004-08-11 16:16:45.000000000 -0600
@@ -107,11 +107,11 @@
 
 install: install-local
 install-local: libclicktool.a install-include
-	$(mkinstalldirs) $(libdir)
-	$(INSTALL) libclicktool.a $(libdir)/libclicktool.a
+	$(mkinstalldirs) $(DESTDIR)/$(libdir)
+	$(INSTALL) libclicktool.a $(DESTDIR)/$(libdir)/libclicktool.a
 install-include:
-	$(mkinstalldirs) $(toolincludedir)
-	$(INSTALL_DATA_IF_CHANGED) $(srcdir)/*.hh $(toolincludedir)
+	$(mkinstalldirs) $(DESTDIR)/$(toolincludedir)
+	$(INSTALL_DATA_IF_CHANGED) $(srcdir)/*.hh $(DESTDIR)/$(toolincludedir)
 uninstall:
 	/bin/rm -f $(libdir)/libclicktool.a
 	cd $(srcdir); for i in *.hh; do /bin/rm -f $(toolincludedir)/$$i; done
diff -ur click-1.4.1-orig/userlevel/Makefile.in click-1.4.1/userlevel/Makefile.in
--- click-1.4.1-orig/userlevel/Makefile.in	2004-07-02 10:46:57.000000000 -0600
+++ click-1.4.1/userlevel/Makefile.in	2004-08-11 16:13:57.000000000 -0600
@@ -118,11 +118,11 @@
 endif
 
 install: $(INSTALLPROGS) install-lib
-	$(mkinstalldirs) $(bindir)
-	for i in $(INSTALLPROGS); do $(INSTALL) $$i $(bindir)/$$i; done
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
+	for i in $(INSTALLPROGS); do $(INSTALL) $$i $(DESTDIR)/$(bindir)/$$i; done
 install-lib: $(INSTALLLIBS)
-	$(mkinstalldirs) $(libdir)
-	for i in $(INSTALLLIBS); do $(INSTALL) $$i $(libdir)/$$i; done
+	$(mkinstalldirs) $(DESTDIR)/$(libdir)
+	for i in $(INSTALLLIBS); do $(INSTALL) $$i $(DESTDIR)/$(libdir)/$$i; done
 uninstall:
 	for i in $(INSTALLLIBS); do rm -f $(libdir)/$$i; done
 	for i in $(INSTALLPROGS); do rm -f $(bindir)/$$i; done
